### PR TITLE
[extractor/teamcoco] Fix extractor

### LINKED ISSUE #6339: Trying to download video from Teamcoco gives 403
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2023.02.17** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

US

### Provide a description that is worded well enough to be understood

Trying to download video from Teamcoco gives:

```
yt-dlp "https://teamcoco.com/video/conan-sona-matt-take-a-hearing-test"
[Teamcoco] Extracting URL: https://teamcoco.com/video/conan-sona-matt-take-a-hearing-test
[Teamcoco] video/conan-sona-matt-take-a-hearing-test: Downloading JSON metadata
ERROR: [Teamcoco] video/conan-sona-matt-take-a-hearing-test: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

By inspecting the web page as its playing, I can get the m3u8 stream and yt-dlp will download the video fine:

```
yt-dlp "https://d3uqdomqytryhw.cloudfront.net/~sw/cdn/public/121317/bf696/hls/CONAF_Segments11723_Hearing%20Test_v3_1920X1080_PRLT_v1--master_960x540_2000k.m3u8"
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://teamcoco.com/video/conan-sona-matt-take-a-hearing-test']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.02.17 [a0a7c01] (pip)
[debug] Python 3.11.1 (CPython x86_64 64bit) - Linux-6.1.11-200.fc37.x86_64-x86_64-with-glibc2.36 (OpenSSL 3.0.8 7 Feb 2023, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.2 (setts), ffprobe 5.1.2
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.02.17, Current version: 2023.02.17
yt-dlp is up to date (2023.02.17)
[Teamcoco] Extracting URL: https://teamcoco.com/video/conan-sona-matt-take-a-hearing-test
[Teamcoco] video/conan-sona-matt-take-a-hearing-test: Downloading JSON metadata
ERROR: [Teamcoco] video/conan-sona-matt-take-a-hearing-test: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 693, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/teamcoco.py", line 110, in _real_extract
    response = self._graphql_call('''{
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/teamcoco.py", line 100, in _graphql_call
    return self._download_json(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1047, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1011, in download_handle
    res = self._download_webpage_handle(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/adobepass.py", line 1366, in _download_webpage_handle
    return super(AdobePassIE, self)._download_webpage_handle(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 881, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 838, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 820, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/langds/.local/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 3726, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/urllib/request.py", line 525, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/urllib/request.py", line 634, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/urllib/request.py", line 563, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/urllib/request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib64/python3.11/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

### PR DESCRIPTION
**Pull request by bashonly**
The Teamcoco extractor needed a complete overhaul:
 - some content from teamcoco.com has been moved to the new site conanclassic.com (added new extractor `ConanClassic`)
 - teamcoco.com no longer uses GraphQL or any API calls
 - conanclassic.com uses a "legacy" version of the graphQL that is slightly different

Closes #6339

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/teamcoco.py [hunk 1]
```diff
@@ -1,57 +1,108 @@
...
+        'low': (480, 272),
+        'sd': (640, 360),
+        'hd': (1280, 720),
+        'uhd': (1920, 1080),
+    }
+
+    def _get_formats_and_subtitles(self, info, video_id):
+        formats, subtitles = [], {}
+
+        for src in traverse_obj(info, ('src', ...), expected_type=dict):
```
Comment (pukkandan) [inline yt_dlp/extractor/teamcoco.py:31]: FYI:
```suggestion
        for src in traverse_obj(info, ('src', ..., {dict})):
```
is the recommended way now

File: yt_dlp/extractor/teamcoco.py [hunk 2]
```diff
@@ -76,126 +124,157 @@ class TeamcocoIE(TurnerBaseIE):
...
-        return self._download_json(
-            'https://teamcoco.com/graphql', object_id, data=json.dumps({
-                'query': query_template % (find_object, object_id)
-            }).encode(), headers={
-                'Content-Type': 'application/json',
-            })['data'][find_object]
 
     def _real_extract(self, url):
-        display_id = self._match_id(url)
+        display_id = self._match_id(url).replace('/', '_')
```
Comment (pukkandan) [inline yt_dlp/extractor/teamcoco.py:131]: display_id can have `/`. You don't have to change this if this is ur preference though

File: yt_dlp/extractor/teamcoco.py [hunk 3]
```diff
@@ -76,126 +124,157 @@ class TeamcocoIE(TurnerBaseIE):
...
+        webpage = self._download_webpage(url, display_id)
+        data = self._search_nextjs_data(webpage, display_id)['props']['pageProps']['pageData']
+        info = merge_dicts(*[traverse_obj(
+            data, ('blocks', lambda _, v: v['name'] == name, 'props'),
+            get_all=False, expected_type=dict, default={}) for name in ('meta-tags', 'video-player', 'video-info')])
```
Comment (pukkandan) [inline yt_dlp/extractor/teamcoco.py:136]:
```suggestion
        info = merge_dicts(*traverse_obj(data, ('blocks',
            lambda _, v: v['name'] in ('meta-tags', 'video-player', 'video-info'),
            'props', {dict}), default={}))
```
(may need re-formatting)

File: yt_dlp/extractor/teamcoco.py [hunk 4]
```diff
@@ -76,126 +124,157 @@ class TeamcocoIE(TurnerBaseIE):
...
-      %s
-    }
-  }
-}''' % self._RECORD_TEMPL, 'Record', child['id'])
-        else:
-            record = response['record']
-        video_id = record['id']
+        thumbnail = traverse_obj(
+            info, (('image', 'poster'), {lambda x: urljoin('https://teamcoco.com/', x)}), get_all=False)
+        video_id = thumbnail.rpartition('=')[2] if thumbnail else display_id
```
Comment (pukkandan) [inline yt_dlp/extractor/teamcoco.py:140]: that's a weird way to get video_id

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/teamcoco.py (hunk 4 above)
Comment (bashonly) [reply yt_dlp/extractor/teamcoco.py:140]: not ideal, but with the site change now it's the only way to get the old video_id that users would have in their archives

File: yt_dlp/extractor/teamcoco.py (hunk 2 above)
Comment (bashonly) [reply yt_dlp/extractor/teamcoco.py:131]: can video_id have `/`? display_id used as a fallback if the "real" video_id can't be extracted

Comment (pukkandan) [reply yt_dlp/extractor/teamcoco.py:131]: Technically yes. But I do agree it's a bad idea

**Pull request merged**